tiledb/sm/rest/curl.cc

```diff
@@ -343,6 +324,10 @@ Status Curl::init(
       "rest.curl.buffer_size", &curl_buffer_size_, &found));
   assert(found);
 
+  RETURN_NOT_OK(config_->get<bool>(
+      "rest.curl.retry_errors", &retry_curl_errors_, &found));
+  assert(found);
```

Comment:
We could use `Config::must_find` to lookup the option and throw if it's not available.

```suggestion
  retry_curl_errors_ = config_->get<bool>("rest.curl.retry_errors", Config::must_find);
```